scene/resources/audio_stream_sample.cpp

```diff
@@ -261,7 +261,8 @@ void AudioStreamPlaybackSample::mix(AudioFrame *p_buffer, float p_rate_scale, in
 		sign = -1;
 	}
 
-	float base_rate = AudioServer::get_singleton()->get_mix_rate();
+	float global_rate_scale = AudioServer::get_singleton()->get_global_rate_scale();
+	float base_rate = AudioServer::get_singleton()->get_mix_rate() * global_rate_scale;
```

Comment:
The equation used in the only other place where `get_global_rate_scale()` is used seems a bit more complex:

```cpp
void AudioStreamPlaybackResampled::mix(AudioFrame *p_buffer, float p_rate_scale, int p_frames) {
        float target_rate = AudioServer::get_singleton()->get_mix_rate();
        float global_rate_scale = AudioServer::get_singleton()->get_global_rate_scale();

        uint64_t mix_increment = uint64_t(((get_stream_sampling_rate() * p_rate_scale) / double(target_rate * global_rate_scale)) * double(FP_LEN));
        ...
```

It's also surprising that it's only used there, does it affect Vorbis samples?

Reply:
Yes, before this, only OGG resources where affected. And while the equation on this one is far simpler, it's still correct from my tests.

Reply:
`) / double(target_rate * global_rate_scale)`

I feel I'm forgetting.  What happens here when scale is set to 0.0?

Reply:
From testing, it just goes silent. And yes, I'm aware that it should error out because of a zero division, but it doesn't seem to happen.

Also, if any problems with that should occur, it would happen even without this commit. So if a fix should be applied, it would be in a different PR.

Reply:
Well a float division by zero is not an error, it gives NaN.

Reply:
After this is merged, I can make that `global_rate_scale` must always be above 0 in the 3.2 branch. For master, there's a PR that already covers that: #38547

Reply:
In fact, I think I can separate #38547 into two commits, with one that can be cherry-picked for 3.2.

Reply:
And done! #43333

Reply:
> Well a float division by zero is not an error, it gives NaN.

I see.

Ah right it was addressed in https://github.com/godotengine/godot/pull/38547.

I remember vaguely testing the first most version of these changes, and yeah, it wasn't crashing, but I recall once global scale was set to 0.0, audio would stay silent even after global scale changed back to something positive.  It may have been because of something downstream regarding the NaN value.  (Are these values being used cumulatively later?)

The inversion of the scaling in PR https://github.com/godotengine/godot/pull/38547, I recall that also being a good correction, and it looks to have the benefit of also avoiding dividing into a NaN.  (Unless the target_rate can be made 0.0)

Again, very foggy on it, I remember this inversion fix working out, as the first most PR scaling towards 0.0 was speeding things up.  Haven't used it in a while, and not sure what combination of these commits should be used now.

Also from looking at this inversion fix.  Wouldn't something similar to this line also work here as well on these other equations? 
`uint64_t mix_increment = uint64_t(((get_stream_sampling_rate() * p_rate_scale * global_rate_scale) / double(target_rate)) * double(FP_LEN));`

Wouldn't 0.0 be the legit minimum?  Pausing/muting audio seems like correct behavior for global scale 0.0.  The problem as I distantly remember it, was that originally one couldn't set scaling to 0.0 then back.  Audio was muted permanently from then on.

Only just some thoughts of mine, since I don't understand the audio math that follows, maybe there are other side effects ahead that I'm not able to take into account.

Reply:
> Well a float division by zero is not an error, it gives NaN.


Actually, `float/0` gives +inf or -inf depending on the sign of the float (or of a potential negative 0), or NaN if it's `0.0/0.0`.